azavea/district-builder-dtl-pa
# Add Rollbar reporting to JS, Django, Celery

## Overview

Adds Rollbar error reporting to the frontend as well as Django and Celery.

### Checklist

- [x] I've edited the .env file on S3 to add the production Rollbar tokens there.

### Notes

I disabled Rollbar in development completely. Rollbar's recommended way of doing things is to always have Rollbar enabled but to use different tokens in different environments, but that seemed like it would be more work to configure properly (setting throttling appropriately, disabling Slack notifications and emails, etc.) than just switching it off, so that's what I did.

## Testing Instructions

 * Make the following changes to your files to enable production Rollbar in your development environment:
```
diff --git a/docker-compose.yml b/docker-compose.yml
index 9222730..4c0432a 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -54,6 +54,8 @@ services:
   celery:
     build:
       context: ./django/publicmapping
+    environment:
+      - CELERY_ROLLBAR=True
     env_file:
       - .env
     volumes:
```
In `.env`, set the appropriate values for `ROLLBAR_SERVER_TOKEN` and `ROLLBAR_CLIENT_TOKEN` using the values from the Rollbar token settings page.

Either in `.env` or `settings.py`, set `DEBUG` to `False`.


 * Edit a Django view to add an error, access that view, and confirm that the error appears in the Rollbar console.
 * Edit a Celery task (I recommend `index2plan`) to add an error, and trigger that task (in the `index2plan` case, by uploading a plan CSV). Confirm that the error appears in the Rollbar console.
 * Remove your induced Python errors and then access the app. Using the browser dev tools, run something similar to this code in the console and confirm that an error appears in the Rollbar console.
 * Use the Rollbar console to resolve the issues you made.

Closes #159012512
